spec/controllers/state_file/questions/az_qualifying_organization_contributions_controller_spec.rb

```diff
@@ -0,0 +1,208 @@
+      contribution = create(:az321_contribution, state_file_az_intake: intake)
+
+      get :index
+
+      expect(response).to be_ok
+      expect(response.body).to include(contribution.charity_name)
+    end
+
+    it 'should hide the add another button when three contributions are already created' do
+      create_list(:az321_contribution, 3, state_file_az_intake: intake)
+
+      get :index
+
+      expect(response).to be_ok
+      expect(response).not_to include('Add another contribution')
+    end
+
+    it 'should show the add another button when there are fewer than three contributions' do
+      create(:az321_contribution, state_file_az_intake: intake)
+
+      get :index
+
+      expect(response).to be_ok
+      expect(response.body).to include('Add another contribution')
+    end
+  end
+
+  describe "edit" do
+    let(:contribution) { create(:az321_contribution, state_file_az_intake: intake)}
+
+    it 'should display a form' do
+      get :edit, params: { id: contribution.id }
+
+      expect(response).to be_ok
+      expect(response.body).to include('<form')
+    end
+
+    it 'should render information about the contribution' do
+      get :edit, params: { id: contribution.id }
+
+      expect(response).to be_ok
+      expect(response.body).to include(contribution.charity_name)
+      expect(response.body).to include(contribution.charity_code)
+      expect(response.body).to include(contribution.amount.to_s("F"))
+    end
+  end
+
+  describe "update" do
+    let(:contribution) do
+      create(
+        :az321_contribution,
+        amount: 50,
+        date_of_contribution_year: Rails.configuration.statefile_current_tax_year,
+        date_of_contribution_month: 5,
+        date_of_contribution_day: 12,
+        state_file_az_intake: intake
+      )
+    end
+
+    it 'should update the contribution when valid' do
+      post :update, params: { id: contribution.id, az321_contribution: { amount: 55 } }
+
+      expect(response).to redirect_to(:az_qualifying_organization_contributions)
+
+      contribution.reload
+
+      expect(contribution.amount.to_i).to be(55)
+    end
+
+    it 'should not update the contribution when invalid' do
+      post :update, params: { id: contribution.id, az321_contribution: { amount: 0 } }
+
+      expect(response).not_to redirect_to(:az_qualifying_organization_contributions)
+
+      contribution.reload
+
+      expect(contribution.amount.to_i).to be(50)
+    end
+
+    it 'should populate the date correctly' do
+      post :update,
+           params: {
+             id: contribution.id,
+             az321_contribution: {
+               date_of_contribution_month: 6,
+               date_of_contribution_day: 16
+             }
+           }
+
+      expect(response).to redirect_to(:az_qualifying_organization_contributions)
+
+      contribution.reload
+
+      expect(contribution.date_of_contribution.to_s).to eql("#{Rails.configuration.statefile_current_tax_year}-06-16")
```

Comment:
oh wow! just had to google `eq` vs `eql`. would this not work with just `eq`? they're both just strings right?

Reply:
Will change